units/log_output_tests.cc

```diff
@@ -0,0 +1,156 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cassert>
+#include <cstddef>
+#include <cstdio>
+#include <fstream>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "debug_logging.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using ::testing::ElementsAre;
+
+namespace performancelayers {
+namespace {
+constexpr unsigned kLogsPerThread = 1337;
+constexpr unsigned kThreadCount = 100;
+
+// An abstraction over the output. It writes the incoming string to the output.
+class LogOutput {
+ public:
+  virtual ~LogOutput() = default;
+
+  virtual void Flush() = 0;
+
+  virtual void LogLine(const std::string &line) = 0;
```

Comment:
nit: I think it would be better to take `std::string_view` as the argument so that we don't force strings to be null-terminated and avoid allocations for string literals. I don't think it matters much though, so feel free to keep this as-is.